logviewer/src/main/java/org/togetherjava/logwatcher/views/usermanagement/UserManagement.java

```diff
@@ -0,0 +1,84 @@
+package org.togetherjava.logwatcher.views.usermanagement;
+
+import com.vaadin.flow.component.orderedlayout.VerticalLayout;
+import com.vaadin.flow.router.PageTitle;
+import com.vaadin.flow.router.Route;
+import org.apache.logging.log4j.LogManager;
+import org.togetherjava.logwatcher.accesscontrol.AllowedRoles;
+import org.togetherjava.logwatcher.accesscontrol.Role;
+import org.togetherjava.logwatcher.constants.UserConstants;
+import org.togetherjava.logwatcher.entities.User;
+import org.togetherjava.logwatcher.users.UserRepository;
+import org.togetherjava.logwatcher.util.NotificationUtils;
+import org.togetherjava.logwatcher.views.MainLayout;
+import org.vaadin.crudui.crud.CrudOperation;
+import org.vaadin.crudui.crud.impl.GridCrud;
+import org.vaadin.crudui.form.CrudFormFactory;
+import org.vaadin.crudui.form.impl.field.provider.CheckBoxGroupProvider;
+
+import javax.annotation.security.PermitAll;
+import java.util.Arrays;
+
+/**
+ * The User-Management View in the Browser
+ */
+
+@PageTitle("UserManagement")
+@Route(value = "user", layout = MainLayout.class)
+@AllowedRoles(roles = {Role.ADMIN})
+@PermitAll
+public class UserManagement extends VerticalLayout {
+
+    private final transient UserRepository repo;
+
+    public UserManagement(UserRepository repository) {
+        this.repo = repository;
+
+        // This is the table you see
+        GridCrud<User> grid = new GridCrud<>(User.class);
+        grid.setOperations(this.repo::findAll, this::onUpdate, this::onUpdate, this::onRemove);
+        grid.getGrid().removeColumnByKey(UserConstants.FIELD_ID);
+
+        final CrudFormFactory<User> factory = grid.getCrudFormFactory();
+        factory.setVisibleProperties(CrudOperation.UPDATE, UserConstants.FIELD_NAME,
+                UserConstants.FIELD_ROLES);
+        factory.setVisibleProperties(CrudOperation.ADD, UserConstants.FIELD_DISCORD_ID,
+                UserConstants.FIELD_NAME, UserConstants.FIELD_ROLES, UserConstants.FIELD_USERNAME);
+        factory.setFieldProvider(UserConstants.FIELD_ROLES,
+                new CheckBoxGroupProvider<>(Arrays.asList(Role.values())));
+        add(grid);
+    }
+
+    /**
+     * Removes the given User from the Repository
+     *
+     * @param user User to remove
+     */
+    private void onRemove(User user) {
+        try {
+            this.repo.delete(user);
+        } catch (RuntimeException e) {
```

Comment:
Is this actually on a top level entry point of the application? If not, you should not catch everything.

Reply:
Changed it to the more specific `IllegalArgumentException`